lib/tasks_private/instance_types.rake

```diff
@@ -1,215 +1,191 @@
+    data_dir.mkpath
+    out_file = data_dir.join('aws_instance_types.yml')
 
-  def format_instance_miq(i)
-    {}.tap do |o|
-      # x1.16large manually fixed to x1.16xlarge
-      # i3.16large manually fixed to i3.16xlarge
-      %i(family instance_store_size instance_store_volumes vpc_only enhanced_networking).each do |key|
-        o[key] = i[key]
+    # weird cache logging issue workaround
+    I18n.backend = I18n.backend.backend unless Rails.initialized?
+
+    LOGGER = Logger.new(STDOUT)
+    LOGGER.info("NOTE: use 'aws:cache:clear' task to clear the cache")
+    AwsProductsDataCollector.cache.logger = LOGGER
+
+    def log_data(header, data, level: :warn)
+      return if data.empty?
+      severity = Logger.const_get(level.to_s.upcase)
+      LOGGER.log(severity) do
+        lines = []
+        lines << header
+        lines << (defined?(AwesomePrint) ? data.ai : JSON.pretty_generate(data))
+        lines.join("\n")
       end
-      o[:memory] = i[:memory].to_f.gigabyte
-      o[:name] = i[:instance_type]
-      o[:description] = i[:pretty_name]
-      o[:vcpu] = i[:vCPU]
-      o[:ebs_optimized_available] = i[:ebs_optimized]
-      o[:instance_store_volumes] = i.fetch_path(:storage, :devices).to_i
-      o[:instance_store_size] = (o[:instance_store_volumes] * i.fetch_path(:storage, :size).to_f).gigabyte
-      o[:instance_store_ssd] = i.fetch_path(:storage, :ssd)
-      o[:architecture] = i[:arch].map(&:to_sym).sort
-      o[:virtualization_type] = i[:linux_virtualization_types].map(&:downcase).map(&:to_sym).sort.map { |v| v == :pv ? :paravirtual : v }
-      o[:network_performance] =
-        case i[:network_performance]
-        when '10 Gigabit', '20 Gigabit', '25 Gigabit'
-          :very_high
-        when ''
-          :unknown
-        else
-          i[:network_performance].downcase.tr(' ', '_').to_sym
-        end
-      o[:ebs_only] = i[:storage].nil? # see https://github.com/ManageIQ/manageiq/issues/741#issuecomment-57353290
-      o[:physical_processor] = i[:physical_processor].nil? ? '' : i[:physical_processor].chomp('*')
-      o[:processor_clock_speed] = i[:clock_speed_ghz]
-      o[:intel_aes_ni] = true # looks like only deprecated types dont support it, so we assume all new types support it
-      #  https://aws.amazon.com/ec2/instance-types/#intel
-      o[:intel_avx] = i[:intel_avx]
-      o[:intel_avx2] = i[:intel_avx2]
-      o[:intel_turbo] = i[:intel_turbo]
-      o[:cluster_networking] = cluster_networking?(i[:instance_type])
     end
-  end
 
-  available = {}
-  previous = {}
-  instances = YAML.safe_load(File.open(File.join(__dir__, 'instance_types_data/instances.json')).read)
-  instances.each do |i|
-    i.deep_symbolize_keys!
-    miq_format = format_instance_miq(i)
-    if i[:generation] == 'current'
-      available[i[:instance_type]] = miq_format
-    else
-      previous[i[:instance_type]] = miq_format
+    # prevent global namespace and memory pollution
+    def isolated
+      yield
```

Comment:
Seems this method is no longer necessary, unless you have some future plans for it?